frame/contracts/src/exec.rs

```diff
@@ -903,11 +905,19 @@ where
 			// Every non delegate call or instantiate also optionally transfers the balance.
 			self.initial_transfer()?;
 
+			#[cfg(feature = "unsafe-debug")]
+			let code_hash = executable.code_hash().clone();
+			#[cfg(feature = "unsafe-debug")]
+			T::Debug::before_call(&code_hash, entry_point, &input_data);
+
 			// Call into the Wasm blob.
 			let output = executable
-				.execute(self, &entry_point, input_data)
+				.execute(self, &entry_point, input_data.clone())
 				.map_err(|e| ExecError { error: e.error, origin: ErrorOrigin::Callee })?;
 
+			#[cfg(feature = "unsafe-debug")]
+			T::Debug::after_call(&code_hash, entry_point, &input_data, &output);
```

Comment:
We can move the input vector since we don't need it anymore. In case the debugger wants to store the data.

```suggestion
			T::Debug::after_call(&code_hash, entry_point, input_clone, &output);
```

Reply:
:white_check_mark: